Dashboard repayment progress

erictshen opened the pull request:
Fixes #119 

Added repayment progress svg and associated stats to project view.

Screenshot below:

![screen shot 2015-06-06 at 9 55 43 pm](https://cloud.githubusercontent.com/assets/2114482/8022675/dc6b8fa8-0c96-11e5-879c-4d44e7993a7f.png)

erictshen commented:
Just creating the pull request now so that feedback can be provided.

Still TODO:
Mobilify
Clean up code some more
Finalize the design (could definitely look better)

noahsark769 commented on revolv/project/models.py:
```diff
@@ -396,6 +397,15 @@ def amount_repaid(self):
         return self.adminrepayment_set.aggregate(models.Sum('amount'))["amount__sum"] or 0.0
 
     @property
+    def total_amount_to_be_repaid(self):
```
Oh nice, I didn't even think that this would be different that the funding goal. Nice catch!

noahsark769 commented on revolv/static/dashboard.scss:
```diff
@@ -399,6 +399,87 @@ $sidebar-icon-gray: #999;
+                position: absolute;
+
+                .svg-graphics-container {
+                // this contains the formatting for the white outside circle - this works
```
Random: this should be indented one level. Also the "this-works" part is pretty useless in terms of explanation, so we should probably get rid of it or expand on the explanation more

noahsark769 commented on revolv/static/dashboard.scss:
```diff
@@ -399,6 +399,87 @@ $sidebar-icon-gray: #999;
+                    position: absolute;
+                    left: 50%;
+                    top: 50%;
+                    transform: translate(-50%, -50%);
```
Can we make a mixin for the prefixing for this?

erictshen replied:
Not fully sure how to do a mixin for scss. And also, a mixin for the circle on the project page and the one in the dashboard?

noahsark769 replied:
Oh, I just meant for the prefixes (make one mixin that handles the transform as well as the -webkit-transform, -moz-transform, etc). Actually now that I look at it, there's already a mixin for transform there! [See here](https://github.com/calblueprint/revolv/blob/master/revolv/static/mixins.scss#L27)

noahsark769 commented on revolv/static/dashboard.scss:
```diff
@@ -399,6 +399,87 @@ $sidebar-icon-gray: #999;
+                    -webkit-transform: translate(-50%, -50%);
+                }
+
+                .project-badge-line, .project-badge-circle {
```
Maybe `repyament-badge` is a better class name here, in order to differentiate it from the other partial circle svg that will be on the funding tab?

erictshen replied:
Good call!

noahsark769 commented on revolv/static/dashboard.scss:
```diff
@@ -655,7 +736,7 @@ $dark-gray: #555555;
                 .explanation, .explanation-header, .choice-header {
                     // this much nesting is necessary because if it is not nested in this far, it does not take precedence
                     width: 75%;
-                    margin-left: 12.5%; 
```
cleanup++

noahsark769 commented on revolv/static/js/base/dashboard.js:
```diff
@@ -48,7 +48,7 @@ $(document).ready(function () {
             },
         });
-    })
+    });
```
dammit @sameerav haha

noahsark769 commented on revolv/static/js/base/dashboard.js:
```diff
@@ -67,7 +67,7 @@ $(document).ready(function () {
     function getCookie(name) {
         var cookieValue = null;
-        if (document.cookie && document.cookie != '') {
+        if (document.cookie && document.cookie !== '') {
```
@sameerav you really need to do `v ssh -c "cd /vagrant && precommit-install"` lol

noahsark769 commented on revolv/static/js/base/dashboard.js:
```diff
@@ -101,10 +101,15 @@ $(document).ready(function () {
         $(".dashboard-data-link.active").removeClass("active");
         $(this).addClass("active");
+        // deletes repayment progress for the current project
+        deleteCurrentCircles();
```
This is in another file, but it's not immediately obvious, so someone reading this might think it was defined in this file, which would be confusing. In order to mitigate this, can we put all the exported stuff from the other file in one object? Like `dashboard.deleteCurrentCircles()` instead?

erictshen replied:
Good call!

noahsark769 commented on revolv/static/js/base/dashboard_project.js:
```diff
@@ -0,0 +1,103 @@
+/**
+ * Resizes the iframe.
+ */
+var resizeIframe = function() {
```
Do we have an iframe on the dashboard?

erictshen replied:
Guess not, and deleted it. A carry over from the project page.

noahsark769 commented on revolv/static/js/base/dashboard_project.js:
```diff
@@ -0,0 +1,103 @@
+            .attr("stroke-width", "3")
+            .attr("fill", "white")
+            .attr("r", outsideCircleRadius);
+            // stroke="black" stroke-width="3"
```
Can delete instead of commenting out

noahsark769 commented on revolv/static/js/base/dashboard_project.js:
```diff
@@ -0,0 +1,103 @@
+            // .attr("r", radius + $(document).width() * 0.03 + "")
+            // .attr("fill", "white");
+
+        if (radius > 100 ){
```
One space between `)` and  `{`

noahsark769 commented on revolv/templates/base/partials/dashboard_project.html:
```diff
@@ -32,6 +32,9 @@
             <dd class="tab-title"><a href="#updates">Updates</a></dd>
         {% endif %}
+        {% if project.is_completed %}
+            <dd class="tab-title"><a href="#repayment-{{project.pk}}">Repayment</a></dd>
```
I think this should be auto-active if the project is completed!

noahsark769 commented:
@erictshen code looks good to me so far, just a couple minor style issues. As for the CSS of the page, I like the circle a lot, but I think the gray outline should be as light as possible, it seems very dark now. For the styles of the table, do you think you can copy the grayish table styles from @sameerav's mockup in #108?

noahsark769 commented:
That weird paypal test is still failing codeship, so don't worry about it for now I guess :P

erictshen commented:
Cleaned up some code and updated the PR! I will try and get some more work in over the week, but I can't guarantee that I'll be able to put work into this until next weekend.

TODO's from code comments : 
Positioning the table better, other design stuff
Mobilify
Comment some more

noahsark769 commented:
Cool cool!

erictshen commented:
Updated styles of design. TODO still : Mobilify (I have no idea how to, and would appreciate some guidance on how to)
Add more comments

![screen shot 2015-06-14 at 10 06 15 pm](https://cloud.githubusercontent.com/assets/2114482/8153354/9d276fbe-12e1-11e5-9488-9c98e1b96241.png)

noahsark769 commented:
Sweet! Can we decrease the size of the text? It looks cramped up into the sides of the grid right now, other than that styles look good I think!

As for mobilifying, are you putting those elements in foundation rows right now? You could just make them take up more columns on mobile, which would push them onto the next line. I don't think any of the other styles need to change other than the fact that the circle and the table have to be on different lines when it gets small enough

erictshen commented:
Not sure why the build is failing. Lmk if you think this is good though.

noahsark769 commented:
@erictshen can i get a screencap?

erictshen commented:
![screen shot 2015-06-16 at 9 14 42 pm](https://cloud.githubusercontent.com/assets/2114482/8199568/d8be3fe4-146c-11e5-83cb-048e23096bcf.png)
![Uploading Screen Shot 2015-06-16 at 9.15.11 PM.png…]()

erictshen commented:
![screen shot 2015-06-16 at 9 15 11 pm](https://cloud.githubusercontent.com/assets/2114482/8199571/e272449a-146c-11e5-82c8-c90f8f093f51.png)

noahsark769 commented:
looks good! Can you take the font size down even a little more? It seems like it's still too big when compared to the latest mockup in #108.

As for the build..anyone try pulling in master?

noahsark769 commented:
Oops maybe* not anyone lol

noahsark769 commented:
Not sure if pulling master will work, worth a shot though

erictshen commented:
![screen shot 2015-06-16 at 9 55 44 pm](https://cloud.githubusercontent.com/assets/2114482/8199952/9a5cc972-1472-11e5-8fe3-ed20aba344d8.png)
![screen shot 2015-06-16 at 9 55 57 pm](https://cloud.githubusercontent.com/assets/2114482/8199953/9d47869a-1472-11e5-8279-52b5695c87d6.png)

noahsark769 commented:
looks great! The build miraculously recovered also, so I'll merge this now. Feel free to get started on another priority 1 issue if you can!

erictshen commented:
Thanks! Build passed b/c of merging with master. Will probably get an svg for Funding Progress on the dashboard next, but not until the weekend.

noahsark769 commented:
Oh awesome! Sounds great, just be sure to assign yourself to an issue as always. I think Vivek and Sameera are working on some of those also. Thanks man!
